deploy/scripts/TLSSettings.ps1

```diff
@@ -0,0 +1,738 @@
+    Write-Log -Message "Creating key: $($CipherSuiteRegKey) "  -Logfile $logLocation -Severity Information
+} 
+ 
+$val = (Get-Item -Path $CipherSuiteRegKey -ErrorAction SilentlyContinue).GetValue("Functions", $null)
+Write-Log -Message "Previous cipher suite value: $val  "  -Logfile $logLocation -Severity Information
+Write-Log -Message "New cipher suite value     : $cipherorder  "  -Logfile $logLocation -Severity Information		 
+	   
+if ($val -ne $cipherorder) 
+{ 
+    Write-Log -Message "Cipher suite order needs to be updated. "  -Logfile $logLocation -Severity Information
+    Write-Host "The original cipher suite order needs to be updated", `n, $val 
+    Set-ItemProperty -Path $CipherSuiteRegKey -Name Functions -Value $cipherorder 
+    Write-Log -Message "Cipher suite value was updated. "  -Logfile $logLocation -Severity Information
+    $reboot = $True 
+}
+else
+{
+    Write-Log -Message "Cipher suite order does not need to be updated. "  -Logfile $logLocation -Severity Information
+	Write-Log -Message "Cipher suite value was not updated as there was no change. " -Logfile $logLocation -Severity Information
+}
+	   	
+#****************************** CHECK THE FIPS SETTING WHICH IMPACTS RDP'S ALLOWED CIPHERS **************************
+#Check for FipsSettings
+Write-Log -Message "Checking to see if reg keys exist and if MinEncryptionLevel is set to 4"  -Logfile $logLocation -Severity Information
+$result = Test-RegistryValueForFipsSettings 
+$reboot = $reboot -or $result
+	
+ 
+#************************************** REBOOT **************************************
+
+if ($RebootIfRequired)  
+{
+    Write-Log -Message "You set the RebootIfRequired flag to true. If changes are made, the system will reboot "  -Logfile $logLocation -Severity Information
+    # If any settings were changed, reboot 
+    If ($reboot) 
+    { 
+        Write-Log -Message "Rebooting now... "  -Logfile $logLocation -Severity Information
+        Write-Log -Message "Using this command: shutdown.exe /r /t 5 /c ""Crypto settings changed"" /f /d p:2:4 "  -Logfile $logLocation -Severity Information
+        Write-Host "Rebooting now..." 
+        shutdown.exe /r /t 5 /c "Crypto settings changed" /f /d p:2:4 
+    }
+    Else 
+    { 
+        Write-Host "Nothing get updated."
+        Write-Log -Message "Nothing get updated. "  -Logfile $logLocation -Severity Information
+    }  
+}
+else
+{
+
+    Write-Log -Message "You set the RebootIfRequired flag to false. If changes are made, the system will NOT reboot "  -Logfile $logLocation -Severity Information
+    Write-Log -Message "No changes will take effect until a reboot has been completed. "  -Logfile $logLocation -Severity Information
+    Write-Log -Message "Script does not include a reboot by design" -Logfile $logLocation -Severity Information
+}
+Write-Log -Message "========== End of logging for a script execution =========="  -Logfile $logLocation -Severity Information
+# SIG # Begin signature block
```

Comment:
what is the purpose of this signature block? looks commented out to me

Reply:
Left by mistake. Removed